File: app/src/main/java/com/antyzero/smoksmog/ui/screen/about/AboutActivity.java
package com.antyzero.smoksmog.ui.screen.about;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.widget.TextView;

import com.antyzero.smoksmog.R;
import com.antyzero.smoksmog.SmokSmogApplication;
import com.antyzero.smoksmog.logger.Logger;
import com.antyzero.smoksmog.ui.BaseActivity;
import com.antyzero.smoksmog.ui.screen.ActivityModule;

import javax.inject.Inject;

import butterknife.Bind;
import butterknife.ButterKnife;

public class AboutActivity extends BaseActivity {

    private static final String TAG = "AboutActivity";

    @Inject
    Logger logger;

    @Bind( R.id.textView )
    TextView textView;
    @Bind( R.id.textViewVersionName )
    TextView textViewVersionName;
    @Bind( R.id.toolbar )
    Toolbar toolbar;

    @Override
    protected void onCreate( Bundle savedInstanceState ) {
        super.onCreate( savedInstanceState );
        SmokSmogApplication.get( this ).getAppComponent().plus( new ActivityModule( this ) ).inject( this );
        setContentView( R.layout.dialog_info_about );
        ButterKnife.bind( this );

        textView.setText( Html.fromHtml( getString( R.string.about ) ) );
        textView.setMovementMethod( LinkMovementMethod.getInstance() );

        try {
            PackageInfo packageInfo = getPackageManager().getPackageInfo( getPackageName(), 0 );
            textViewVersionName.setText( getString( R.string.version_name_and_code,
                    packageInfo.versionName,
                    packageInfo.versionCode ) );
        } catch ( Exception e ) {
            logger.i( TAG, "Problem with obtaining version", e );
        }

        setSupportActionBar( toolbar );
        //noinspection ConstantConditions
        getSupportActionBar().setDisplayHomeAsUpEnabled( true );
    }

    public static void start( Context context ) {
        context.startActivity( intent( context ) );
    }

    public static Intent intent( Context context ) {
        return new Intent( context, AboutActivity.class );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/about/AboutActivity.kt
package com.antyzero.smoksmog.ui.screen.about

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.method.LinkMovementMethod
import com.antyzero.smoksmog.R
import com.antyzero.smoksmog.SmokSmogApplication
import com.antyzero.smoksmog.dsl.compatFromHtml
import com.antyzero.smoksmog.ui.BaseActivity
import com.antyzero.smoksmog.ui.screen.ActivityModule
import kotlinx.android.synthetic.main.dialog_info_about.*
import smoksmog.logger.Logger
import javax.inject.Inject

class AboutActivity : BaseActivity() {

    @Inject lateinit var logger: Logger

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        SmokSmogApplication[this].appComponent.plus(ActivityModule(this)).inject(this)
        setContentView(R.layout.dialog_info_about)

        textView.compatFromHtml(R.string.about)
        textView.movementMethod = LinkMovementMethod.getInstance()

        try {
            val packageInfo = packageManager.getPackageInfo(packageName, 0)
            textViewVersionName.text = getString(R.string.version_name_and_code,
                    packageInfo.versionName,
                    packageInfo.versionCode)
        } catch (e: Exception) {
            logger.i(TAG, "Problem with obtaining version", e)
        }

        setSupportActionBar(toolbar)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    companion object {

        private val TAG = "AboutActivity"

        fun start(context: Context) {
            context.startActivity(intent(context))
        }

        fun intent(context: Context): Intent {
            return Intent(context, AboutActivity::class.java)
        }
    }
}
